### ISSUE TITLE
Tiny SSD: A Tiny Single-shot Detection Deep Convolutional Neural Network for Real-time Embedded Object Detection

### ISSUE THREAD

**Issue by Swall0w** [Swall0w/papers#653]
Alexander Wong, Mohammad Javad Shafiee, Francis Li, Brendan Chwyl

Object detection is a major challenge in computer vision, involving both object classification and object localization within a scene. While deep neural networks have been shown in recent years to yield very powerful techniques for tackling the challenge of object detection, one of the biggest challenges with enabling such object detection networks for widespread deployment on embedded devices is high computational and memory requirements. Recently, there has been an increasing focus in exploring small deep neural network architectures for object detection that are more suitable for embedded devices, such as Tiny YOLO and SqueezeDet. Inspired by the efficiency of the Fire microarchitecture introduced in SqueezeNet and the object detection performance of the single-shot detection macroarchitecture introduced in SSD, this paper introduces Tiny SSD, a single-shot detection deep convolutional neural network for real-time embedded object detection that is composed of a highly optimized, non-uniform Fire sub-network stack and a non-uniform sub-network stack of highly optimized SSD-based auxiliary convolutional feature layers designed specifically to minimize model size while maintaining object detection performance. The resulting Tiny SSD possess a model size of 2.3MB (~26X smaller than Tiny YOLO) while still achieving an mAP of 61.3% on VOC 2007 (~4.2% higher than Tiny YOLO). These experimental results show that very small deep neural network architectures can be designed for real-time object detection that are well-suited for embedded scenarios.

https://arxiv.org/abs/1802.06488

**Comment by vasoyajignesh7** [comment]
Hi,

I am trying to train model for COCO 2017 dataset. mbox_loss reduces up to 8.0 within 1k iterations. But loss is not decreasing less than 8.0 even after more than 1 lac iteration run. I have also tried different hyper parameter values like batch size, base_lr, gamma, lr_policy (mutistep, Poly) and optimization (RMSprop, sgd) but not success yet.

Great if you can suggest suitable solution to achieve best training loss/ mbox_loss. 
For your reference attaching training log file and plot which is achieved best so far after multiple trials.

Hope you will be able to provide solution soon.

[coco_train_log_Trial_2.log](https://github.com/Swall0w/papers/files/4856278/coco_train_log_Trial_2.log)
![train_Trial_2](https://user-images.githubusercontent.com/20473126/86214888-72490580-bb99-11ea-8d59-2cb2b5aaea2c.png)


- Jignesh Vasoya
